sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlParser.g

```diff
@@ -2216,6 +2216,8 @@ regularBody[boolean topLevel]
 selectStatement[boolean topLevel]
    :
    (
+   (
+   LPAREN
```

Comment:
Because in the following we need to access some of the sub clauses such as selectClause, clusterByClause, etc., with the syntax seems we can't simply extract the common part as a single rule.